src/System.Private.CoreLib/src/System/Runtime/CompilerServices/jithelpers.cs

```diff
@@ -94,9 +94,12 @@ internal class RawData
         public byte Data;
     }
 
-    internal class RawSzArrayData
+    internal class RawArrayData
     {
-        public IntPtr Count; // Array._numComponents padded to IntPtr
+        public uint Length; // Array._numComponents padded to IntPtr
+#if BIT64
+        public uint Padding;
+#endif
         public byte Data;
     }
```

Comment:
Maybe worth moving it to the shared? Along with new Length and LongLength.

Reply:
Maybe eventually. There is a lot of runtime specific quirks.

Does Mono use the same memory layout for array as CoreCLR?

Reply:
Not the same layout but similar one. Length/Data is exposed to the managed code in the same way.